packages/core/src/state-management/repository.ts

```diff
@@ -985,15 +985,19 @@ export class Repository {
   updates$(init: StreamState): Observable<StreamState> {
     return new Observable<StreamState>((subscriber) => {
       const id = new StreamID(init.type, init.log[0].cid)
-      this.fromMemoryOrStore(id).then((found) => {
-        const state$ = found || new RunningState(init, false)
-        this.inmemory.endure(id.toString(), state$)
-        state$.subscribe(subscriber).add(() => {
-          if (state$.observers.length === 0) {
-            this.inmemory.free(id.toString())
-          }
+      this.fromMemoryOrStore(id)
+        .then((found) => {
+          const state$ = found || new RunningState(init, false)
+          this.inmemory.endure(id.toString(), state$)
+          state$.subscribe(subscriber).add(() => {
+            if (state$.observers.length === 0) {
+              this.inmemory.free(id.toString())
+            }
+          })
+        })
+        .catch((error) => {
+          this.logger.err(`An error occurred in updates$ for StreamID ${id}: ${error}`)
```

Comment:
Shouldn't this update propagate to the Observable @stbrody?

Reply:
Sorry I meant propagating the error to the observable consumer, otherwise here it is swallowed by the `catch()` handler.

Reply:
yeah this definitely looks like a behavior change at the least.  I think we probably do indeed want the error to propagate to the consumer.  I'll tag in @ukstv for his take as well though.

@samika98, if you don't add the .catch here, does the new lint rule complain?  That would be surprising to me.

Reply:
Yes the new lint rule does complain if I don't add the catch here. However, if we do want to silent it we can.

Reply:
@samika98 Indeed this is a change. If `catch` is required, you could add something like:
```
promise.catch(error => { subscriber.error(error) })
```

Reply:
Added the fix suggested